src/components/Composer/index.js

```diff
@@ -86,387 +88,349 @@ const propTypes = {
-            const pastedHTML = event.clipboardData.getData(TEXT_HTML);
+            // If paste contains HTML
+            if (types.includes(TEXT_HTML)) {
+                const pastedHTML = event.clipboardData.getData(TEXT_HTML);
 
-            const domparser = new DOMParser();
-            const embeddedImages = domparser.parseFromString(pastedHTML, TEXT_HTML).images;
+                const domparser = new DOMParser();
+                const embeddedImages = domparser.parseFromString(pastedHTML, TEXT_HTML).images;
 
-            // Exclude parsing img tags in the HTML, as fetching the image via fetch triggers a connect-src Content-Security-Policy error.
-            if (embeddedImages.length > 0 && embeddedImages[0].src) {
-                // If HTML has emoji, then treat this as plain text.
-                if (embeddedImages[0].dataset && embeddedImages[0].dataset.stringifyType === 'emoji') {
-                    this.handlePastePlainText(event);
-                    return;
+                // Exclude parsing img tags in the HTML, as fetching the image via fetch triggers a connect-src Content-Security-Policy error.
+                if (embeddedImages.length > 0 && embeddedImages[0].src) {
+                    // If HTML has emoji, then treat this as plain text.
+                    if (embeddedImages[0].dataset && embeddedImages[0].dataset.stringifyType === 'emoji') {
+                        handlePastePlainText(event);
+                        return;
+                    }
                 }
+                handlePastedHTML(pastedHTML);
+                return;
             }
-
-            this.handlePastedHTML(pastedHTML);
-            return;
-        }
-
-        this.handlePastePlainText(event);
-    }
+            handlePastePlainText(event);
+        },
+        [onPasteFile, translate, paste, handlePastedHTML, checkComposerVisibility],
+    );
 
     /**
      * Manually scrolls the text input, then prevents the event from being passed up to the parent.
      * @param {Object} event native Event
      */
-    handleWheel(event) {
+    const handleWheel = useCallback((event) => {
         if (event.target !== document.activeElement) {
             return;
         }
 
-        this.textInput.scrollTop += event.deltaY;
+        textInput.current.scrollTop += event.deltaY;
         event.preventDefault();
         event.stopPropagation();
-    }
+    }, []);
 
-    /**
-     * We want to call updateNumberOfLines only when the parent doesn't provide value in props
-     * as updateNumberOfLines is already being called when value changes in componentDidUpdate
-     */
-    shouldCallUpdateNumberOfLines() {
-        if (!_.isEmpty(this.props.value)) {
+    useEffect(() => {
+        if (textInput.current === null) {
             return;
         }
 
-        this.updateNumberOfLines();
-    }
+        // we reset the height to 0 to get the correct scrollHeight
+        textInput.current.style.height = 0;
```

Comment:
Setting the height to 0 and subsequently to 'auto' appears to be a workaround. Could you help me understand why the composer started jumping after the functional component refactor? I didn't observe this behaviour in the main branch.

Reply:
Yes, it's a workaround instead `this.setState(({numberOfLines: 1}) => {this.setState({numberOfLines: number})})`. With FC I wasn't able to mirror this stuff.
When exactly it jumps for you?

Reply:
I see, I have referred to the commit 'fix jumping'

Reply:
I see, back then, I hadn't used `textInput.current.style.height = 0;` and tried to do this as it was initially with state swapping. It was the issue. I've returned updateNumberOfLines, as you mentioned.